src/Gui/SoFCColorGradient.cpp

```diff
@@ -110,20 +110,24 @@ void SoFCColorGradient::setMarkerLabel(const SoMFString& label)
 
 void SoFCColorGradient::setViewportSize(const SbVec2s& size)
 {
-    // don't know why the parameter range isn't between [-1,+1]
-    float fRatio = ((float)size[0]) / ((float)size[1]);
+    float fRatio = static_cast<float>(size[0]) / static_cast<float>(size[1]);
     float fMinX =  4.0f, fMaxX = 4.5f;
     float fMinY = -4.0f, fMaxY = 4.0f;
 
     if (fRatio > 1.0f) {
-        fMinX = 4.0f * fRatio;
+        fMinX = 5.0f * fRatio;

```

Comment:
same here

Reply:
The explanation above. The y values of the normalized coordinates are in the range [-5, 5] because of the `height = 10`. Because the ratio > 1 the x values are in the range [-5*ratio, +5*ratio]. In the past 4.0 was used that worked quite well for all of my monitors and until now there were never any complaints :)